### PR TITLE
Add notes about do not validate rw's generated columns on source

### LINKED ISSUE #2112: Document: feat: do not valid rw's generated column on source
This issue tracks the documentation update needed for the merged PR #16522.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/16522
Source PR Merged At: 2024-04-28T10:29:27Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/16522

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2112#event-12651850748

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

File: docs/sql/commands/sql-create-source.md [hunk 1]
```diff
@@ -58,6 +58,12 @@ FORMAT upsert ENCODE AVRO (
 );
 ```
 
+:::note
+
+The generated column is created in RisingWave and will not be accessed through the external connector. Therefore, do not validate RisingWave's generated columns on the source.
+
+:::
```
Comment (st1page) [inline docs/sql/commands/sql-create-source.md]: Sorry for the misunderstanding. The term "valid" here refers to the logic within the RisingWave kernel, which we do not need to explain to the users. 
```suggestion
:::note

The generated column is created in RisingWave and will not be accessed through the external connector. Therefore, If the external upstream system has a schema, it does not need to include the generated column within the table's schema in the external system.

:::
```
We only need to mention in the release notes that after version 1.9, RW supports defining generated columns in the tables with the CDC connector.

File: docs/sql/query-syntax/query-syntax-generated-columns.md [hunk 2]
```diff
@@ -9,13 +9,19 @@ title: Generated columns
...
 ```sql
 CREATE TABLE t1 (v1 int AS v2-1, v2 int, v3 int AS v2+1);
 ```
 
-Note that a generation expression cannot reference another generated column.
+:::note
+
++ A generation expression cannot reference another generated column.
+
++ The generated column is created in RisingWave and will not be accessed through the external connector. Therefore, do not validate RisingWave's generated columns on the source.
```
Comment (st1page) [inline docs/sql/query-syntax/query-syntax-generated-columns.md]: ditto

File: docs/sql/commands/sql-create-source.md (hunk 1 above)
Comment (ShanlanLi) [reply docs/sql/commands/sql-create-source.md]: > We only need to mention in the release notes that after version 1.9, RW supports defining generated columns in the tables with the CDC connector.

@st1page Thanks! I'll tell @emile-00  to add this to the release note. Then do I still need to add the note to our doc:

> The generated column is created in RisingWave and will not be accessed through the external connector. Therefore, If the external upstream system has a schema, it does not need to include the generated column within the table's schema in the external system.

Comment (st1page) [reply docs/sql/commands/sql-create-source.md]: > Then do I still need to add the note to our doc

I think yes. we can notify it to user.

**Review by WanYixian — APPROVED** [review]

**Review by st1page — APPROVED** [review]

**Pull request merged**